1. 공장에서는 시제품의 농도(%)가 60이라고 주장하며 품질관리팀에서 10개의 샘플을 뽑았다. 유의수준 5%에서 다음을 검정하시오.
> 데이터 : 52, 50, 62, 75, 26, 45, 62, 35, 57, 14

(1) 연구가설, 귀무가설을 작성하시오.
(2) 유효한 샘플의 수를 계산하시오.
(3) 검정통계량을 구하고 연구가설 채택 여부를 작성하시오.

(1) 연구가설, 귀무가설을 작성하시오.
* 연구가설 : 시제품의 농도는 60%가 아니다.
* 귀무가설 : 시제품의 농도는 60%이다.

(2) 유효한 샘플의 수를 계산하시오.
* 유효한 샘플의 수는 최소 표본크기를 계산하여 설정할 수 있다.
* 최소 표본크기의 계산은 허용오차식을 바탕으로 한다.
* 허용할 수 있는 오차한계와 신뢰수준이 주어졌을 때, 샘플로부터 추정한 표준편차를 사용해 표본의 크기를 구할 수 있다.
* 오차한계를 5, 신뢰수준을 0.05로 설정한 뒤 샘플의 표준편차를 구한다.

In [12]:
import pandas as pd

x=[52, 50, 62, 75, 26, 45, 62, 35, 57, 14]

import numpy as np
moe = 5
a=0.05
std = np.std(x)
print(std)

# 유효한 샘플의 수 구하기(from scipy)
import scipy
import math
n_sample = (scipy.stats.t.ppf(q=0.05, df=9)*std/5)**2
print('유효한 샘플의 수는 ', math.ceil(n_sample), '이상 입니다.')

17.434448657758008
유효한 샘플의 수는  41 이상 입니다.


(3) 검정통계량을 구하고 연구가설 채택 여부를 작성하시오.
* Wilcoxon의 부호검정을 검정방법으로 선정한다. 이 검정방식은 모집단이 정규성을 만족하지 못할 때 사용하는 검정이며, 비교대상들 사이의 차이를 부호와 크기에 대한 정보로 바꾸어 이를 토대로 분포를 비교하는 방식이다.

In [14]:
from scipy import stats
import pandas as pd
stats.wilcoxon(pd.Series(x)-60) # 편차

WilcoxonResult(statistic=9.5, pvalue=0.064453125)

#### * p-value가 0.06으로 유의수준 0.05보다 큰 것을 확인할 수 있다.
#### * 즉, 유의수준 0.05 하에서 귀무가설을 채택한다. 따라서 시제품의 농도는 60%이라고 판단할 수 있다.

2. 코로나 데이터 시계열 분석(feat.덴드로그램) TBU

3. 사회과학, 자연과학, 공학 세 개 학과의 평점조사표를 보고, 학과와 성적이 관계있는지 검정하시오.
> 데이터 : 세 개 학과의 평점조사표

|구분|사회과학|자연과학|공학|
|:--:|:--:|:--:|:--:|
|3.5~4.5|16|12|18|
|2.5~3.5|30|20|13|
|1.5~2.5|12|3|14|


(1) 귀무가설, 대립가설을 작성하시오.
(2) 학과와 성적이 독립일 때, 기댓값을 구하시오.
(3) 검정통계량을 구하고 연구가설 채택 여부를 작성하시오.

(1) 귀무가설, 대립가설을 작성하시오.
* 귀무가설 : 성적과 학과 간에는 관련성이 없다. 즉, 독립이다.
* 대립가설 : 성적과 학과 간에는 관련성이 있다. 즉, 독립이 아니다.

이때, 카이제곱 검정을 검정방식으로 채택한다.
* 카이제곱 검정은 관찰된 빈도와 기대되는 빈도가 의미 있게 다른지의 여부를 검정하기 위해 사용하는 방법이다.
* 명목척도 자료 분석에 이용되며, 자료가 빈도로 주어졌을 때 사용한다.
* 카이제곱 검정은 목적에 따라 3가지로 사용될 수 있다.
    * 적합도 검정 : 관찰된 비율 값이 기댓값과 같은지 조사(어떤 모집단의 표본이 그 모집단을 대표하는지 검정)
    * 동질성 검정 : 두 집단의 분포가 동일한지 검정
    * 독립성 검정 : Contigency Table에서 두 개 이상의 변수가 서로 독립인지 검정

In [23]:
# df 구성
import pandas as pd
society = [16, 30, 12]
nature = [12, 20, 3]
machine = [18, 13, 14]
df = pd.DataFrame({'sociery':society,
                   'nature' : nature,
                   'machine' : machine},
                  index = ['3.5~4.5', '2.5~3.5', '1.5~2.5'])

df

sociery  nature  machine
3.5~4.5       16      12       18
2.5~3.5       30      20       13
1.5~2.5       12       3       14

#### * 카이제곱 검정 중 독립성 검정 선택

In [24]:
from scipy import stats
stats.chi2_contingency(observed=df)

(10.199441509990177,
 0.03719883770303157,
 4,
 array([[19.33333333, 11.66666667, 15.        ],
        [26.47826087, 15.97826087, 20.54347826],
        [12.1884058 ,  7.35507246,  9.45652174]]))

#### * 카이제곰 검정통계량은 10.199이고, p-value는 0.037이다.
#### * 이에 따라 유의수준 0.05 하에서 귀무가설을 기각하고 대립가설을 채택한다.
#### 즉, 성적과 학과 간에는 관련성이 있는 것으로 판단할 수 있다.